client/state/ui/actions/set-sites.js

```diff
@@ -6,7 +6,7 @@ import 'calypso/state/ui/init';
  * Returns an action object to be used in signalling that a site has been set
  * as selected.
  *
- * @param {number} siteId Site ID
+ * @param {number | null} siteId Site ID
```

Comment:
I was concerned about this, but then I realized we are calling `setSelectedSiteId( null );` in other parts in the app as well, so it makes sense to open up this type and make it explicit that we can set it to null.